pymeasure/instruments/activetechnologies/AWG401x.py

```diff
@@ -0,0 +1,1145 @@
+            Generator""",
+            **kwargs
+        )
+
+    def beep(self):
+        """Causes a system beep."""
+        self.write("SYST:BEEP")
+
+    def save(self, position):
+        """Save the actual configuration in memory.
+
+        Parameters
+        ----------
+        position : int
+            Instrument save position [0,4]
+
+        Raises
+        ------
+        ValueError
+            If position is outside permitted limit [0,4].
+        """
+
+        if position >= 0 or position <= 4:
+            self.write(f"*SAV {position}")
+        else:
+            raise ValueError("position value outside permitted range [0,4]")
+
+    def load(self, position):
+        """Load the actual configuration in memory.
+
+        Parameters
+        ----------
+        position : int
+            Instrument load position [0,4]
+
+        Raises
+        ------
+        ValueError
+            If position is outside permitted limit [0,4].
+        """
+
+        if position >= 0 or position <= 4:
+            self.write(f"*RCL {position}")
+        else:
+            raise ValueError("position value outside permitted range [0,4]")
+
+    def waitLast(self):
+        """Wait for last operation completition"""
+
+        self.write("*WAI")
+
+
+class AWG401x_AFG(AWG401x_base):
+    """Represents the Active Technologies AWG-401x Arbitrary Waveform Generator
+    in AFG mode.
+
+    .. code-block:: python
+
+        wfg = AWG401x_AFG("TCPIP::192.168.0.123::INSTR")
+
+        wfg.reset()                     # Reset the instrument at default state
+
+        wfg.ch[1].shape = "SINUSOID"    # Sets a sine waveform on CH1
+        wfg.ch[1].frequency = 4.7e3     # Sets the frequency to 4.7 kHz on CH1
+        wfg.ch[1].amplitude = 1         # Set amplitude of 1 V on CH1
+        wfg.ch[1].offset = 0            # Set the amplitude to 0 V on CH1
+        wfg.ch[1].enabled = True        # Enables the CH1
+
+        wfg.ch[2].shape = "SQUARE"      # Sets a square waveform on CH2
+        wfg.ch[2].frequency = 100e6     # Sets the frequency to 100 MHz on CH2
+        wfg.ch[2].amplitude = 0.5         # Set amplitude of 0.5 V on CH2
+        wfg.ch[2].offset = 0            # Set the amplitude to 0 V on CH2
+        wfg.ch[2].enabled = True        # Enables the CH2
+
+        wfg.enabled = True              # Enable output of waveform generator
+        wfg.beep()                      # "beep"
+
+        print(wfg.check_errors())       # Get the error queue
+
+    """
+
+    enabled = Instrument.control(
+        "AFGControl:STATus?", "AFGControl:%s",
+        """A boolean property that enables the generation of signals.""",
+        validator=strict_discrete_set,
+        values={True: "START", False: "STOP"},
+        map_values=True,
+        get_process=lambda v: "START" if v == 1 else ("STOP" if v == 0 else v)
+    )
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(adapter)
```

Comment:
kwargs are silently discarded. Is this intentional?

Reply:
Fixed